models/spring-ai-ollama/src/test/java/org/springframework/ai/ollama/api/tool/OllamaApiToolFunctionCallIT.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.ai.ollama.api.OllamaApi.Message.Role;
+import org.springframework.ai.ollama.api.OllamaApi.Message.ToolCall;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.ollama.OllamaContainer;
+
+/**
+ * @author Christian Tzolov
+ */
+@Disabled("For manual smoke testing only.")
+@Testcontainers
+public class OllamaApiToolFunctionCallIT {
+
+	private static String MODEL = "mistral";
+
+	private static final Logger logger = LoggerFactory.getLogger(OllamaApiToolFunctionCallIT.class);
+
+	MockWeatherService weatherService = new MockWeatherService();
+
+	@Container
+	static OllamaContainer ollamaContainer = new OllamaContainer("ollama/ollama:0.2.8");
+
+	static String baseUrl = "http://localhost:11434";
+
+	@BeforeAll
+	public static void beforeAll() throws IOException, InterruptedException {
+		logger.info("Start pulling the '" + MODEL + " ' generative ... would take several minutes ...");
+		ollamaContainer.execInContainer("ollama", "pull", MODEL);
+		logger.info(MODEL + " pulling competed!");
+
+		baseUrl = "http://" + ollamaContainer.getHost() + ":" + ollamaContainer.getMappedPort(11434);
+	}
+
+	@SuppressWarnings("null")
+	@Test
+	public void toolFunctionCall() {
+
+		OllamaApi completionApi = new OllamaApi(baseUrl);
+
+		// Step 1: send the conversation and available functions to the model
+		var message = Message.builder(Role.USER)
+			// .withContent("What's the weather like in San Francisco, Tokyo, and Paris?
+			// Perform multiple function calls for each location.")
+			.withContent("What's the weather like in San Francisco, Tokyo, and Paris?")
+			.build();
+
+		var functionTool = new OllamaApi.ChatRequest.Tool(new OllamaApi.ChatRequest.Tool.Function("getCurrentWeather",
+				"Get the weather in location. Return temperature in Celsius.", ModelOptionsUtils.jsonToMap("""
+						{
+							"type": "object",
+							"properties": {
+								"location": {
+									"type": "string",
+									"description": "The city and state e.g. San Francisco, CA"
+								},
+								"unit": {
+									"type": "string",
+									"enum": ["C", "F"]
+								}
+							},
+							"required": ["location", "unit"]
+						}
+						""")));
+
+		List<Message> messages = new ArrayList<>(List.of(message));
+
+		OllamaApi.ChatRequest chatCompletionRequest = OllamaApi.ChatRequest.builder(MODEL)
+			.withMessages(messages)
+			.withTools(List.of(functionTool))
+			.build();
+
+		ChatResponse chatCompletion = completionApi.chat(chatCompletionRequest);
+
+		assertThat(chatCompletion).isNotNull();
+		assertThat(chatCompletion.message()).isNotNull();
+
+		Message responseMessage = chatCompletion.message();
+
+		assertThat(responseMessage.role()).isEqualTo(Role.ASSISTANT);
+		assertThat(responseMessage.toolCalls()).isNotNull();
+
+		// Check if the model wanted to call a function
+		if (responseMessage.toolCalls() != null) {
```

Comment:
This check shouldn't be needed since the previous assertion makes sure the tool calls are not null.

Reply:
removing